sessionCheckFrame.src.js

```diff
@@ -0,0 +1,93 @@
+(function () {
+    "use strict";
+    /**
+     * This code is designed to run in the context of a window (or frame)
+     * that has been loaded as the redirect_uri target of an OIDC grant. Parameters
+     * passed to this window may be provided in the hash fragment values or in the
+     * query string; in either case, they will be provided as normal key1=value&key2=value
+     * entries. Note that it is expected that there should never be any "access_token"
+     * values present in either form of the parameters.
+     *
+     * For more details on the hash fragment query string format see :
+     * https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse
+     *
+     * If you are using the "id_token" response type, the code expects there to be
+     * these values in sessionStorage prior to handling the authentication response:
+     *
+     * - "sessionCheckNonce" - This should be set during the authentication request, and it must
+     *                         match the value found in the "nonce" claim of the id_token
+     *
+     * - "sessionCheckSubject" - This is the user that is currently logged-in to the RP. It
+     *                           must match the "sub" claim of the id_token, otherwise it is
+     *                           assumed that a different user has logged into the OP and the RP
+     *                           session is therefore invalid.
```

Comment:
Shouldn't "sessionCheckIdToken" be mentioned here as an alternative to "sessionCheckSubject"?

Reply:
In retrospect, I think I'll just simplify the whole thing and remove the sessionCheckIdToken entirely. If people want subject validation, they can provide the subject themselves.